src/compute/curve.rs

```diff
@@ -1,24 +1,34 @@
 use crate::compute::ElementP2;
 use curve25519_dalek::ristretto::RistrettoPoint;
 
-pub trait Curve {
+pub trait SwCurveConfig: ark_ec::short_weierstrass::SWCurveConfig {
     fn curve_id() -> u32;
 }
 
-impl Curve for RistrettoPoint {
+impl SwCurveConfig for ark_bls12_381::g1::Config {
     fn curve_id() -> u32 {
-        blitzar_sys::SXT_CURVE_RISTRETTO255
+        blitzar_sys::SXT_CURVE_BLS_381
     }
 }
 
-impl Curve for ElementP2<ark_bls12_381::g1::Config> {
+impl SwCurveConfig for ark_bn254::g1::Config {
     fn curve_id() -> u32 {
-        blitzar_sys::SXT_CURVE_BLS_381
+        blitzar_sys::SXT_CURVE_BN_254
     }
 }
 
-impl Curve for ElementP2<ark_bn254::g1::Config> {
+pub trait Curve {
+    fn curve_id() -> u32;
```

Comment:
You could use an associated constant instead:
```suggestion
    const CURVE_ID: u32;
```